semanticdb/semanticdb3/guide.md

```diff
@@ -0,0 +1,727 @@
+  * `Symbols` contains information about definitions in the source
+    file, including modifiers, signatures, etc.
+
+    For example, `_empty_.Test.main(Array). => method main: (args: Array[String]): Unit`
+    says that `main` is a method with one parameter of type `Array[String]`.
+    Further lines of the printout establish that `Array` in that type
+    refers to `scala.Array#`, etc.
+  * `Occurrences` contains a list of identifiers from the source file with
+    their line/column-based positions and unique identifiers pointing to
+    corresponding definitions resolved by the compiler.
+
+    For example, `[2:4..2:11): println => scala.Predef.println(Any).` says that
+    the identifier `println` on line 3 (zero-based numbering scheme!) refers
+    to the one-argument `println` overload from `scala.Predef`.
+
+## What is SemanticDB good for?
+
+SemanticDB decouples producers and consumers of semantic information about
+programs and establishes [a rigorous specification](semanticdb3.md) of the
+interchange format.
+
+Thanks to that, SemanticDB-based tools like [Scalafix](#scalafix),
+[Metadoc](#metadoc) and [Metals](#metals) don't need to know about compiler
+internals and can work with any compiler that supports SemanticDB.
+This demonstrably improves developer experience, portability and scalability.
+Next-generation semantic tools at Twitter are based on SemanticDB.
+
+For more information about the SemanticDB vision, check out our talks:
+  * [Semantic Tooling at Twitter](https://www.youtube.com/watch?v=4yqDFsdKciA)
+    (June 2017) by Eugene Burmako & Stu Hood.
+  * [SemanticDB for Scala developer tools](https://t.co/Z080F6JxyQ?amp=1)
+    (April 2018) by Ólafur Páll Geirsson.
+  * [How We Built Tools That Scale to Millions of Lines of Code](https://na.scaladays.org/schedule/how-we-built-tools-that-scale-to-millions-of-lines-of-code)
+    (June 2018) by Eugene Burmako.
+
+## Producing SemanticDB
+
+### Scalac compiler plugin
+
+The `semanticdb-scalac` compiler plugin injects itself immediately after the
+`typer` phase of the Scala compiler and then harvests and dumps semantic
+information from Scalac in SemanticDB format.
+
+```
+Usage: scalac -Xplugin:path/to.jar -Yrangepos [<pluginOption> ...] [<scalacOption> ...] [<sourceFile> ...]
+```
+
+The compiler plugin supports the following options that can
+be passed through Scalac in the form of `-P:semanticdb:<option>:<value>`
+
+<table>
+  <tr>
+    <td width="310px">Option</td>
+    <td>Value</td>
+    <td>Explanation</td>
+    <td>Default</td>
+  </tr>
+  <tr>
+    <td><code>-P:semanticdb:sourceroot:&lt;value&gt;</code></td>
+    <td>Absolute or relative path</td>
+    <td>
+      Used to relativize source file paths into
+      <code>TextDocument.uri</code>.
+    </td>
+    <td>Current working directory</td>
+  </tr>
+  <tr>
+    <td><code>-P:semanticdb:mode:&lt;value&gt;</code></td>
+    <td>
+        <code>fat</code>,<br/>
+        <code>slim</code>
```

Comment:
Missing trailing comma

Reply:
Oops, I didn't mean for trailing commas to be a thing.